streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -294,29 +295,19 @@ public void resume() {
         }
     }
 
-    /**
-     * @throws TaskMigratedException if committing offsets failed (non-EOS)
-     *                               or if the task producer got fenced (EOS)
-     */
     @Override
-    public void commit() {
+    public Map<TopicPartition, OffsetAndMetadata> prepareCommit() {
         switch (state()) {
             case RUNNING:
             case RESTORING:
-                commitState();
-
-                // this is an optimization for non-EOS only
-                if (eosDisabled) {
-                    stateMgr.checkpoint(checkpointableOffsets());
-                }
-
-                log.info("Committed");
+            case SUSPENDED:
```

Comment:
We need to allow committing in `SUSPENDED` state now as we first suspend all tasks and than commit. Cf. `TaskManager#handleRevocation()`